main.go

```diff
@@ -198,7 +204,7 @@ func run(c *cli.Context) error {
 	if len(filename) == 0 {
 		log.Info("No filename specified. Creating default")
 
-		filename = "archive.tar"
+		filename = fmt.Sprintf("archive.%s", c.String("archive_type"))
```

Comment:
I feel like you should be checking that archive type is valid before using it everywhere.

So I would check that its valid and then use that value instead of `c.String`. This is similar to the other code hanging out.